client/vaultclient/vaultclient.go

```diff
@@ -256,6 +264,59 @@ func (c *vaultClient) DeriveToken(alloc *structs.Allocation, taskNames []string)
 	return tokens, nil
 }
 
+func (c *vaultClient) DeriveTokenWithJWT(ctx context.Context, reqs map[string]JWTLoginRequest) (map[string]string, error) {
+	if !c.config.IsEnabled() {
+		return nil, fmt.Errorf("vault client not enabled")
+	}
+	if !c.isRunning() {
+		return nil, fmt.Errorf("vault client is not running")
+	}
+
+	c.lock.Lock()
+	defer c.unlockAndUnset()
+
+	// Make sure the login request is not passing any token.
+	c.client.SetToken("")
+
+	var g multierror.Group
+	var tokensSync sync.Map
+
+	for k, v := range reqs {
+		name := k
+		req := v
+		g.Go(func() error {
+			s, err := c.client.Logical().WriteWithContext(ctx, "auth/jwt/login",
```

Comment:
Hi, I was looking into this and seeing this, i thought that in vault currently [only one jwks_url can be set](https://developer.hashicorp.com/vault/api-docs/auth/jwt#configure), so if that auth backend (auth/<backend>/login) is already used by another jwt backend this won't work. Can we set the <backend> configurable in the client configuration?

Reply:
That's a good catch @jorgemarey. From https://developer.hashicorp.com/vault/api-docs/auth/jwt:

> This documentation assumes the plugin method is mounted at the /auth/jwt path in Vault. Since it is possible to enable auth methods at any location, please update your API calls accordingly.

Reply:
> please update your API calls accordingly.

This feels very targeted to me 😄 

Thanks @jorgemarey, I would've 100% missed that!

@tgross I think we need a new field in the `vault` agent config to allow this value to be customized.

Reply:
> @tgross I think we need a new field in the vault agent config to allow this value to be customized.

Agreed. I can pick that up.